code/modules/projectiles/updated_projectiles/guns/specialist.dm

```diff
@@ -153,11 +153,17 @@
 	current_mag = /obj/item/ammo_magazine/rifle/m4ra
 	force = 16
 	attachable_allowed = list(
+						/obj/item/attachable/heavy_barrel,
+						/obj/item/attachable/extended_barrel,
 						/obj/item/attachable/suppressor,
 						/obj/item/attachable/verticalgrip,
 						/obj/item/attachable/angledgrip,
 						/obj/item/attachable/bipod,
-						/obj/item/attachable/compensator)
+						/obj/item/attachable/compensator,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/attached_gun/shotgun,
```

Comment:
Standard question: have to tested to see how those sprites look?

Reply:
EB misaligned, can confirm.

Reply:
Fixed